src/Hpack/Config.hs

```diff
@@ -231,12 +233,12 @@ instance HasFieldNames ExecutableSection
 instance FromJSON ExecutableSection where
   parseJSON = genericParseJSON_
 
-dependenciesFromList :: Maybe (List Dependency) -> Map String DependencyVersion
-dependenciesFromList xs = Map.fromList [(name, version) | Dependency name version <- fromMaybeList xs]
+dependenciesFromList :: Maybe Dependencies -> Dependencies
+dependenciesFromList = fromMaybe mempty
```

Comment:
Arguably, this is not a good name for this function anymore.  I would tend to just inline it.